pymc/logprob/mixture.py

```diff
@@ -300,13 +300,27 @@ def mixture_replace(fgraph, node):
 
     old_mixture_rv = node.default_output()
 
-    mixture_res, join_axis = get_stack_mixture_vars(node)
+    try:
+        mixture_res, join_axis = get_stack_mixture_vars(node)
+    except NotImplementedError:
```

Comment:
This seems like a code smell. Would it be unreasonable to refactor `get_stack_mixture_vars` to not raise this exception?

Reply:
I refactored it to avoid raising the exception